compiler/sempass2.nim

```diff
@@ -1102,7 +1103,8 @@ proc checkRaisesSpec(g: ModuleGraph; spec, real: PNode, msg: string, hints: bool
   if hints:
     for s in 0..<spec.len:
       if not used.contains(s):
-        message(g.config, spec[s].info, hintXDeclaredButNotUsed, renderTree(spec[s]))
+        message(g.config, spec[s].info, hintXCannotRaiseY,
+                "'$1' cannot raise '$2'" % [renderTree(hintsArg), renderTree(spec[s])])
```

Comment:
This won't look good with lambdas, best to just not mention the name of the proc, since the error message will point to the pragma location anyways.

Reply:
I tested
```nim
let a = (proc (x: int) {.raises: [IOError].} = return)
```

and I got
```
/home/idf/fisiere/src/nim-idf/examples/raisins.nim(4, 35) Hint: ':anonymous' cannot raise 'IOError' [XCannotRaiseY]
```
which doesn't look bad, and it points the line aswell